option/option.go

```diff
@@ -269,3 +270,57 @@ type withClientCertSource struct{ s ClientCertSource }
 func (w withClientCertSource) Apply(o *internal.DialSettings) {
 	o.ClientCertSource = w.s
 }
+
+// WithImpersonatedCredentials returns a ClientOption that will impersonate the
+// target service account.
+//
+// In order to impersonate the target service account
+// the base service account must have the Service Account Token Creator role,
+// roles/iam.serviceAccountTokenCreator, on the target service account.
+// See https://cloud.google.com/iam/docs/understanding-service-accounts.
+//
+// Optionally, delegates can be used during impersonation if the base service
+// account lacks the token creator role on the target. When using delegates,
+// each service account must be granted roles/iam.serviceAccountTokenCreator
+// on the next service account in the chain.
+//
+// For example, if a base service account of SA1 is trying to impersonate target
+// service account SA2 while using delegate service accounts DSA1 and DSA2,
+// the following must be true:
+//
+//   1. Base service account SA1 has roles/iam.serviceAccountTokenCreator on
+//      DSA1.
+//   2. DSA1 has roles/iam.serviceAccountTokenCreator on DSA2.
+//   3. DSA2 has roles/iam.serviceAccountTokenCreator on target SA2.
+//
+// The resulting impersonated credential will either have the default scopes of
+// the client being instantiating or the scopes from WithScopes if provided.
+// Scopes are required for creating impersonated credentials, so if this option
+// is used while not using a NewClient/NewService function, WithScopes must also
+// be explicitly passed in as well.
+//
+// If the base credential is an authorized user and not a service account, or if
+// the option WithQuotaProject is set, the target service account must have a
+// role that grants the serviceusage.services.use permission such as
+// roles/serviceusage.serviceUsageConsumer.
+//
+// This is an EXPERIMENTAL API and may be changed or removed in the future.
+func WithImpersonatedCredentials(target string, delegates ...string) ClientOption {
```

Comment:
Most of the time, WithX means you're passing in an X.

In this case, we're not passing in impersonated credentials (which would be a *google.Credentials type, I think) but rather named delegates.

`WithImpersonatedDelegate" perhaps?

"WithImpersonationTarget" ?

Sorry, just riffing. I don't think those are particularly good names, but that's all I've got right now.

Reply:
Do you like `ImpersonateCredentials`? Just drop the with?

Reply:
Done.